src/app/_components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+
+interface TextInputProps {
+  id: string;
+  name: string;
+  placeholder?: string;
+  value?: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  errorMessage?: string;
+  successMessage?: string;
+}
+
+const TextInput: React.FC<TextInputProps> = ({
+  id,
+  name,
+  placeholder,
+  value,
+  onChange,
+  onFocus,
+  onBlur,
+  errorMessage,
+  successMessage,
+}) => {
+  const isError = Boolean(errorMessage);
+  const isSuccess = Boolean(successMessage);
+
+  return (
+    <div className="flex flex-col">
+      <input
+        id={id}
+        name={name}
+        type="text"
```

Comment:
I think type should be a prop since there are various types that can affect how input behave like email and password. You could also extend your input to HTML Input to gain the props of that so that your component can then have those props as well. The button I made has an example of this.